cross-modal-search/pods/executors.py

```diff
@@ -11,10 +11,40 @@
 from jina.helloworld.chatbot.my_executors import _norm, _ext_B, _ext_A, _cosine
 
 
+class ReciprocalRankEvaluator(Executor):
+
+    @staticmethod
+    def _evaluate(actual: Sequence[Union[str, int]], desired: Sequence[Union[str, int]], *args, **kwargs) -> float:
+        """
+        Evaluate score as per reciprocal rank metric.
+
+        :param actual: Sequence of sorted document IDs.
+        :param desired: Sequence of sorted relevant document IDs
+            (the first is the most relevant) and the one to be considered.
+        :param args:  Additional positional arguments
+        :param kwargs: Additional keyword arguments
+        :return: Reciprocal rank score
+        """
+        if len(actual) == 0 or len(desired) == 0:
+            return 0.0
+        try:
+            return 1.0 / (actual.index(desired[0]) + 1)
+        except:
+            return 0.0
+
+    @requests
```

Comment:
do we want to evaluate on `index` and `search` time?

Reply:
no, just search time, thx for pointing out

Reply:
although I believe `/search: rank_evaluate` in evaluate.yml also does the job